tensor2tensor/data_generators/text_encoder.py

```diff
@@ -337,31 +355,33 @@ def build_from_token_counts(self,
             starts.append(pos)
             pos += len(self.subtoken_to_subtoken_string(subtoken))
         for start in starts:
-          for end in xrange(start + 1, len(escaped_token) + 1):
```

Comment:
No, `end` is only used as a slice endpoint of `escaped_token` and you never need (or want) its value to go beyond `len(escaped_token)`, which is one past the last index of the string. The last character of the string is `escaped_token[len(escaped_token)-1 : len(escaped_token)]`.